universal_bridge.md

```diff
@@ -0,0 +1,456 @@
+        string name;
+        /// @notice Symbol of the token
+        string symbol;
+        /// @notice Decimals of the token
+        uint8 decimals;
+    }
+```
+
+### Minting CET on the fly
+
+The bridge can mint CETs on demand via a factory
+
+```solidity
+interface ICETFactory {
+    function predictAddress(address l1Asset) external view returns (address predicted);
+    function deployIfAbsent(
+        address l1Asset,
+        uint8 decimals,
+        string calldata name,
+        string calldata symbol,
+        address bridge
+    ) external returns (address deployed);
+}
+
+ICETFactory public cetFactory;
+
+function computeCETAddress(address remoteAsset) internal view returns (address) {
+    return cetFactory.predictAddress(remoteAsset);
+}
+
+function ensureCETAndMint(
+    address remoteAsset,
+    string calldata name,
+    string calldata symbol,
+    uint8 decimals,
+    address to,
+    uint256 amount
+) internal returns (address cet) {
+    // 1) Predict deterministic address
+    address predicted = computeCETAddress(remoteAsset);
+
+    // 2) Deploy if missing (CREATE3-based factory)
+    cet = cetFactory.deployIfAbsent(remoteAsset, decimals, name, symbol, address(this));
+    require(cet == predicted, "CET address mismatch");
+
+    // 3) Mint via bridge-only path
+    IToken(cet).mint(to, amount);
+    return cet;
+}
+```
+
+-------------
+
+###  Mailbox
+
+Mailbox is a container of `Messages` divided into 2 boxes: 
+- `Inbox` that has messages consumed by `Read()` function
+- `Outbox` that has messages pushed into by `Write()` function.
+
+
+``` solidity
+  struct Message {
+    MessageHeader header,
+    bytes payload
+  }
+
+// Header for message. Its hash can serve as the message Identifier.
+    struct MessageHeader {
+        uint256 sessionId; // 16 bytes of version + 240 bytes of value chosen by the user
+        uint256 chainSrc;  // chain ID where the message was sourced from
+        uint256 chainDest; // chain ID of target destination
+        address sender;    // The address on `chainSrc` initiating the message 
+        address receiver;  // The address on `chainDest` receiving the message
+        string label;      // Helps decipher the payload.
+    }    
+
+interface IMailbox {
+    // `sender` writes to the OUTBOX a message to be relayed to `reciever` on `chainDest`
+    function write(
+        Message calldata message
+    ) external;
+
+    // `receiver` reads from the INBOX a message relayed by `sender` from `srcChain`
+    function read(
+        MessageHeader calldata header
+    ) external returns (bytes memory);
+}
+```
+
+
+
+#### SessionID
+
+SessionID is a 240 bits random value that MUST NOT be reused across MessageHeaders with otherwise similar values.
+However, every message in a cross-chain exchange mapping to a single atomic operation must carry the same SessionID.
+
+The first 16 bytes serve as version. Currently the only canonical version is 0.
+
```

Comment:
Great decision. Do you think it makes sense to include here the mechanism for implementing it? like:
```
sessionId = uint256(keccak256(abi.encode(
    VERSION << 240,  // 16-bit version prefix
    sender,
    nonce,
    blockNumber,
    userSalt
)));
```